### ISSUE TITLE
Check if installing ucx-proc is still necessary/helpful

### ISSUE THREAD

**Issue by pentschev** [rapidsai/ucx-py#400]
The [current instruction](https://ucx-py.readthedocs.io/en/latest/quickstart.html#setup) is to install `ucx-proc` for both GPU and CPU. I have verified that when we install `cudatoolkit` as well, this isn't necessary. It would be helpful to double-check whether installing the CPU version (without `cudatoolkit` for non-GPU work) is necessary/helpful.

**Comment by jakirkham** [comment]
Following up on our discussion earlier, I filed issue ( https://github.com/conda-forge/ucx-split-feedstock/issues/66 ) to investigate building GPU bits always and simply not requiring a GPU at runtime.

**Comment by pentschev** [comment]
I have checked this now, and with `strace` and monitoring `nvidia-smi` I was able to verify that when you install `ucx-py`, by default it's going to use CUDA libraries, but once you install `ucx-proc=*=cpu` this is not anymore the case. That means that for CPU we really need `ucx-proc=*=cpu` but `ucx-proc=*=gpu` doesn't seem necessary (behavior is the same with and without it).

I guess overall, there's no need to change anything in the docs at this time. However, the resolution of https://github.com/conda-forge/ucx-split-feedstock/issues/66 would potentially allow us to change this.

@jakirkham please let me know your thoughts on this.

**Comment by jakirkham** [comment]
@mrocklin, if you have time, would you be able to test this out as well? It would be useful to know if the GPU packages just work on CPU-only machines. This would mean we can ship one package and use it everywhere pretty easily 🙂

**Comment by quasiben** [comment]
I tested this out with the following install on an Intel NUC:

> conda create -n ucx -c rapidsai-nightly ucx ucx-py python=3.7 pytest-asyncio

The was able to successfully execute test_send_recv.py:

>  UCXPY_IFNAME=enp0s25 py.test -s -v test_send_recv.py

**Comment by jakirkham** [comment]
Are you getting the GPU build in that case?

**Comment by quasiben** [comment]
Yes, I am picking up cudatoolkit:

> cudatoolkit               10.2.89              hfd86e86_0    defaults

**Comment by quasiben** [comment]
however, on RTD we are getting an error:
> Command killed due to excessive memory consumption

I think this is due to cudatoolkit being quite large -- 500MB+

@jakirkham do you have thoughts on how to proceed .  Should we update the install instructions ?

**Comment by jakirkham** [comment]
Yeah I do. Needed the info you provided above to be sure though.

I think we can do the same thing `openmpi` did recently.

Will work up a PR in conda-forge first and then we can apply in RAPIDS as well.

**Comment by jakirkham** [comment]
One thing we would still need to know though is how UCX responds to being built against different CUDA versions. Have raised as issue ( https://github.com/openucx/ucx/issues/4762 ).

**Comment by quasiben** [comment]
@jakirkham i filed https://github.com/readthedocs/readthedocs.org/issues/6647 to ask RTD for more memory when building

**Comment by jakirkham** [comment]
Thanks for doing that.

I'm hoping we will be able to progress without needing changes on RTD. Just need a bit more info before doing so. 🙂

**Comment by quasiben** [comment]
with latest ucx/ucx-py from rapidsai-night i installed onto a CPU only machine and was unable to load ucp:

```
In [1]: import ucp
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
<ipython-input-1-0e8d48451255> in <module>
----> 1 import ucp

~/miniconda3/envs/ucx-test/lib/python3.7/site-packages/ucp/__init__.py in <module>
      8
      9 from ._version import get_versions as _get_versions
---> 10 from .core import *  # noqa
     11 from .core import get_ucx_version
     12 from .utils import get_address, get_ucxpy_logger  # noqa

~/miniconda3/envs/ucx-test/lib/python3.7/site-packages/ucp/core.py in <module>
     15 import psutil
     16
---> 17 from ._libs import ucx_api
     18 from ._libs.utils import get_buffer_nbytes
     19 from .continuous_ucx_progress import BlockingMode, NonBlockingMode

ImportError: libcuda.so.1: cannot open shared object file: No such file or directory
```

**Comment by jakirkham** [comment]
Yeah we are currently caring around patches that don't allow us to drop this. We can investigate again once we no longer need those patches.

**Comment by pentschev** [comment]
The `ucx-proc` package is not anymore useful and was completely removed from the conda-forge package. This isn't relevant anymore, closing.
